src/ray/core_worker/core_worker.cc

```diff
@@ -419,30 +419,34 @@ Status CoreWorker::Wait(const std::vector<ObjectID> &ids, int num_objects,
         plasma_store_provider_->Wait(plasma_object_ids, num_objects, /*timeout_ms=*/0,
                                      worker_context_.GetCurrentTaskID(), &ready));
   }
-  if (memory_object_ids.size() > 0) {
+  RAY_CHECK(static_cast<int>(ready.size()) <= num_objects);
+  if (memory_object_ids.size() > 0 && static_cast<int>(ready.size()) < num_objects) {
```

Comment:
nit: please reverse the order of these conditions to match the two if statements below

Reply:
(or reverse both of those)